Sources/TuistCore/Graph/ModelExtensions/Target+Core.swift

```diff
@@ -106,8 +106,18 @@ extension Target {
                 .subtracting(excluded)
                 .filter { path in
                     guard let `extension` = path.extension else { return false }
-                    return Target.validSourceExtensions
+
+                    let hasValidSourceExtensions = Target.validSourceExtensions
                         .contains(where: { $0.caseInsensitiveCompare(`extension`) == .orderedSame })
+
+                    if hasValidSourceExtensions {
+                        // Addition check to prevent folders with name like `Foo.Swift` to be considered as source files.
+                        return !FileHandler.shared.isFolder(path)
+                    } else {
```

Comment:
Can't we do this `if` before checking  `Target.validSourceExtensions`, which AFAIIU is useless if the path is a folder?